src/main/java/io/strimzi/ValidatingWebhook.java

```diff
@@ -74,32 +83,35 @@ public AdmissionReview webhook(AdmissionReview review) {
         AdmissionRequest request = review.getRequest();
         ObjectMeta evictionMetadata = extractEvictionMetadata(request);
 
-        if (evictionMetadata != null)   {
-            if (drainZooKeeper && ZOOKEEPER_PATTERN.matcher(evictionMetadata.getName()).matches()
-                    || drainKafka && KAFKA_PATTERN.matcher(evictionMetadata.getName()).matches()) {
-                String name = evictionMetadata.getName();
-                String namespace = evictionMetadata.getNamespace();
-
-                if (namespace == null) {
-                    // Some applications (see https://github.com/strimzi/drain-cleaner/issues/34) might send the eviction
-                    // request without the namespace. In such case, we use the namespace form the AdmissionRequest.
-                    LOG.warn("There is no namespace in the Eviction request - trying to use namespace of the Admission request");
-                    namespace = request.getNamespace();
-                }
-
-                if (name == null || namespace == null) {
-                    LOG.warn("Failed to decode pod name or namespace from the eviction webhook (pod: {}, namespace: {})", name, namespace);
+        if (evictionMetadata != null && evictionMetadata.getNamespace() != null) {
+            String name = evictionMetadata.getName();
+            String namespace = evictionMetadata.getNamespace();
+
+            Pod pod = client.pods().inNamespace(namespace).withName(name).get();
```

Comment:
You should search for the pod only after checking that both name and namespace are not null or blank. Below we also get the namespace from the request in case it is missing from the eviction, but this code is never executed.

After these checks, you can get the pod, match it against our rules and pass it as parameter to `annotatePodForRestart` to avoid doing another API call.